src/libopensc/card-iasecc.c

```diff
@@ -445,16 +447,22 @@ iasecc_oberthur_match(struct sc_card *card)
 {
 	struct sc_context *ctx = card->ctx;
 	unsigned char *hist = card->reader->atr_info.hist_bytes;
+	size_t hist_bytes_len = card->reader->atr_info.hist_bytes_len;
 
 	LOG_FUNC_CALLED(ctx);
 
-	if (*hist != 0x80 || ((*(hist+1)&0xF0) != 0xF0))
-		LOG_FUNC_RETURN(ctx, SC_ERROR_OBJECT_NOT_FOUND);
-
-	sc_log_hex(ctx, "AID in historical_bytes", hist + 2, *(hist+1) & 0x0F);
+	if (*hist == 0x80 && ((*(hist+1)&0xF0) == 0xF0)) {
```

Comment:
Please fix the possible out of bounds access by checking `hist_bytes_len`. (I know, you didn't introduce it!)

Reply:
`&& (*(hist+1) & 0x0F) <= hist_bytes_len - 2`?

Reply:
`if (hist_bytes_len >= 2 && *hist == 0x80 && ((*(hist+1)&0xF0) == 0xF0))`